### ISSUE TITLE
the reward doesn't increase

### ISSUE THREAD

**Issue by ML117554** [alexjameswilliams/KukaHybridVisualServoing#1]
step = 5000: AverageReturn = 7.617593, AverageEpisodeLength = 21.000000
step = 10000: AverageReturn = 7.661649, AverageEpisodeLength = 21.049999
step = 15000: AverageReturn = 7.671409, AverageEpisodeLength = 21.049999
step = 20000: AverageReturn = 7.703858, AverageEpisodeLength = 21.049999
step = 25000: AverageReturn = 7.724912, AverageEpisodeLength = 21.049999
step = 30000: AverageReturn = 7.678084, AverageEpisodeLength = 21.049999
step = 35000: AverageReturn = 7.790011, AverageEpisodeLength = 21.049999
step = 40000: AverageReturn = 7.644900, AverageEpisodeLength = 21.049999
step = 45000: AverageReturn = 7.659567, AverageEpisodeLength = 21.049999
step = 50000: AverageReturn = 7.885606, AverageEpisodeLength = 21.049999
step = 55000: AverageReturn = 7.561671, AverageEpisodeLength = 21.049999
step = 60000: AverageReturn = 7.672000, AverageEpisodeLength = 21.049999
step = 65000: AverageReturn = 7.716365, AverageEpisodeLength = 21.049999
step = 70000: AverageReturn = 7.660107, AverageEpisodeLength = 21.049999
step = 75000: AverageReturn = 7.754662, AverageEpisodeLength = 21.049999
step = 80000: AverageReturn = 7.722183, AverageEpisodeLength = 21.049999
step = 85000: AverageReturn = 7.784870, AverageEpisodeLength = 21.049999
step = 90000: AverageReturn = 7.689379, AverageEpisodeLength = 21.049999
step = 95000: AverageReturn = 7.739729, AverageEpisodeLength = 21.049999
step = 100000: AverageReturn = 7.799435, AverageEpisodeLength = 21.049999

Why hasn't the reward increased when running your repository? Is it because of a version issue with the Python library? Can you give me your requirements.txt

**Comment by alexjameswilliams** [comment]
Hi,

Thanks for taking an interest in the repository.

Unfortunately as this work is now a few years old, I do not have a requirements.txt, or access to the machine this work was originally completed on.

However, if you read the [thesis](https://cronfa.swan.ac.uk/Record/cronfa62598) the work came out of, the versions of key libraries were:
Tensorflow==2.6
Tensorflow Agents==0.9
PyBullet==3.2.0
Comet.ml==3.15

Unfortunately, I was quite inexperienced in Deep RL when conducting this work and was not able to get an agent to train in any meaningful way in the environment (read the thesis for more detail + analysis if interested). I'm not sure why the reward isn't significantly changing. It could be a bug! But it could also just be bad hyperparameter settings or a neural network with insufficient capacity to learn, or a limited exploration strategy.

The main contribution of this repo is the environment itself which is reusable. It would be trivial to adapt the environment for use with an RL library of choice instead. But the script also shows how it can be setup with TF-Agents so you can play around with the settings.

I'm currently researching in a different field for my PhD and don't have any plans to come back to this work anytime soon (though I would like to at some point). If you find any bugs or make any updates to the code then feel free to push to the repository! Alternatively, feel free to drop me an email and we could discuss collaborating to bring the work forward. 

Thanks,
Alex
